src/round.rs

```diff
@@ -0,0 +1,162 @@
+// This is a part of Chrono.
+// See README.md and LICENSE.txt for details.
+
+use Timelike;
+use std::ops::{Add, Sub};
+use oldtime::Duration;
+
+/// Extension trait for subsecond rounding or truncation to a maximum number
+/// of digits. Rounding can be used to decrease the error variance when
+/// serializing/persisting to lower precision. Truncation is the default
+/// behavior in Chrono display formatting.  Either can be used to guarantee
+/// equality (e.g. for testing) when round-tripping through a lower precision
+/// format.
+pub trait SubSecondRound {
+    /// Return a copy rounded to the specified number of subsecond digits. With
+    /// 9 or more digits, self is returned unmodified. Halfway values are
+    /// rounded up (away from zero).
+    fn round(self, digits: u16) -> Self;
+
+    /// Return a copy truncated to the specified number of subsecond
+    /// digits. With 9 or more digits, self is returned unmodified.
+    fn trunc(self, digits: u16) -> Self;
+}
+
+impl<T> SubSecondRound for T
+where T: Timelike + Add<Duration, Output=T> + Sub<Duration, Output=T>
+{
+    fn round(self, digits: u16) -> T {
+        let span = span_for_digits(digits);
+        let rem = self.nanosecond() % span;
+        if rem > 0 {
+            let rev = span - rem;
```

Comment:
In 74ec7aa5 I've renamed variables to  `delta_down` and `delta_up`.  Logicwise, I believe this is optimal, since `delta_up` is used for the up branch.